text/3537-msrv-resolver.md

```diff
@@ -0,0 +1,1276 @@
+
+In some cases, the MSRV-incompatible dependencies might be restricted to `dev-dependencies`.
+Though local development can't be performed with the MSRV,
+the fact that the tests are verifying (on a newer toolchain) that the build/normal dependencies work gives a good amount of confidence that they will work on the MSRV so long as they compile.
+
+Compared to the above workflow, this is likely targeted at just library and tool maintainers as other use cases don't have access to the latest version or they are needing the repo to be compatible with their MSRV.
+
+**Priority 4 because:**
+- ❌ The MSRV has various carve outs, providing an inconsistent experience compared to other packages using other workflows and affecting the quality of the ecosystem
+  - For workspaces with bins, `cargo install --locked` is expected to work with the MSRV but won't
+  - If they use new Cargo features, then `[patch]`ing in a git source for the dependency won't work
+  - For contributors, they must be on an unspecified Rust toolchain version
+- ❌ The caveats involved in this approach (see prior item) would lead to worse documentation which lowers the quality to users
+- ❌ This still leads to stagnation despite being able to use the latest dependencies as they are limited in what they can use from them and they can't use features from the latest Rust toolchain
+- ❌ These library and tool maintainers are absorbing costs from the less common use cases of their dependents
+  - They could shift these costs to those that need old versions by switching to the "Latest MSRV" workflow by allowing their users to backport fixes to prior MSRV releases
+
+**Pain points:**
+
+Like the prior workflow, when developing with the latest toolchain,
+feedback is delayed until CI or an embedded image build process which can be frustrating
+(using too-new dependencies, using too-new Cargo or Rust features, etc).
+
+When `Cargo.lock` is resolved for latest dependencies, independent of MSRV,
+verifying the MSRV becomes difficult as they  must either juggle two lockfiles, keeping them in sync, or use the unstable `-Zminimal-versions`.
+The two lockfile approach also has all of the problems shown earlier in writing the lockfile.
+
+When only keeping MSRV-incompatible `dev-dependencies`,
+one lockfile can be used but it can be difficult to edit the `Cargo.lock` to ensure you get new `dev-dependencies` without infecting other dependency types.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+We are introducing several new concepts
+- A v3 resolver (`package.resolver`) that will prefer packages compatible with your `package.rust-version` over those that aren't
+  - If `package.rust-version` is unset, then your current Rust toolchain version will be used
+  - This resolver version will be the default for the next edition
+  - A `.cargo/config.toml` field will be added to disable this, e.g. for CI
```

Comment:
@djc (moved to better follow the threads)

> I don't see how a config.toml field will make it easier to disable this only in CI?

All config fields are implicitly environment variables which you can set on the individual step or whole job within your CI.

In the reference section, we call out the name of this variable.